WordPress/src/main/java/org/wordpress/android/modules/ApplicationModule.java

```diff
@@ -1,15 +1,40 @@
 package org.wordpress.android.modules;
 
 import android.app.Application;
+import android.arch.lifecycle.ViewModel;
+import android.arch.lifecycle.ViewModelProvider;
 import android.content.Context;
+import android.support.v4.util.ArrayMap;
+
+import org.wordpress.android.viewmodel.PluginBrowserViewModel;
+
+import javax.inject.Provider;
+import javax.inject.Singleton;
 
 import dagger.Binds;
 import dagger.Module;
+import dagger.Provides;
 
-@Module
+@Module(subcomponents = ViewModelSubComponent.class)
 public abstract class ApplicationModule {
     // Expose Application as an injectable context
     @Binds
     abstract Context bindContext(Application application);
+
+    @Provides
+    @Singleton
+    public static ViewModelProvider.Factory provideViewModelFactory(ViewModelSubComponent.Builder subComponentBuilder) {
```

Comment:
How do you feel about moving this to it's own module, maybe called `ViewModelModule`? I might be wrong about this, but it looks like we prefer to keep things in their own module and that's why `ApplicationModule` is empty.

Reply:
If you make this change, can you please remove those extra 2 lines at the end? We have style checks for these in FluxC library, but not here, although I think we are looking to add it.

Reply:
Great point;). Fixed